### ISSUE TITLE
Any features like tensorflow.minimize(loss)

### ISSUE THREAD

**Issue by lincolnBush** [dmlc/mxnet#5272]
Dear all:

I wonder  is there are any features in mxnet which work like tensorflow's minimize loss functions as in tensorflow, it is only needed to write a loss function in python and do not need to define the backward and forward operations. That might be very useful for developers and for mxnet to become more popular. Many thanks.

**Comment by sxjscience** [comment]
@lincolnBush In MXNet, this is achieved via `mx.sym.MakeLoss` http://mxnet.io/api/python/symbol.html#mxnet.symbol.MakeLoss. You can first construct the loss function with some basic symbols and use MakeLoss to get a loss symbol that will pass-back gradient.

**Comment by sxjscience** [comment]
https://github.com/dmlc/mxnet/blob/master/example/ssd/symbol/symbol_vgg16_reduced.py#L135-L136 and https://github.com/dmlc/mxnet/blob/bcd4a33445ae17bf9da064f0e8f9316658ad69c0/example/reinforcement-learning/ddpg/qfuncs.py#L47

**Comment by lincolnBush** [comment]
Thank you very much
